jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/Resource.java

```diff
@@ -175,24 +176,31 @@ public boolean isDirectory()
     }
 
     /**
-     * Time resource was last modified.
+     * The time the resource was last modified.
+     *
      * Equivalent to {@link Files#getLastModifiedTime(Path, LinkOption...)} with the following parameter:
-     * {@link #getPath()} then returning {@link FileTime#toMillis()}.
+     * {@link #getPath()} then returning {@link FileTime#toInstant()}.
      *
-     * @return the last modified time as milliseconds since unix epoch or
-     * 0 if {@link Files#getLastModifiedTime(Path, LinkOption...)} throws {@link IOException}.
+     * @return the last modified time instant, or null if resource doesn't exist or doesn't have a last modified time.
      */
-    public long lastModified()
+    public Instant lastModified()
     {
+        Path path = getPath();
+        if (path == null)
+            return null;
+
+        if (!Files.exists(path))
+            return null;
+
         try
         {
-            FileTime ft = Files.getLastModifiedTime(getPath(), FOLLOW_LINKS);
-            return ft.toMillis();
+            FileTime ft = Files.getLastModifiedTime(path, FOLLOW_LINKS);
+            return ft.toInstant();
         }
         catch (IOException e)
         {
             LOG.trace("IGNORED", e);
-            return 0;
+            return null;
```

Comment:
Return `Instant.EPOCH` instead.

Reply:
null for not a path, and not-exists, but epoch for failure?

Reply:
Done.